Source/Applications/ApplicationArtifactIdentifier.cs

```diff
@@ -55,7 +55,7 @@ public class ApplicationArtifactIdentifier : IApplicationArtifactIdentifier
         /// <inheritdoc/>
         public bool Equals(IApplicationArtifactIdentifier other)
         {
-            if (!Location.Equals(other.Location)) return false;
+            if (Location != other.Location) return false;
 
             if (Application.Name.Value != other.Application.Name.Value) return false;
```

Comment:
This should be, Application.Equals(other.Application)
But that's scary now. Should we just stick with the Name for a while?

@einari